tests/client/test_http_unicode.py

```diff
@@ -0,0 +1,247 @@
+                        "text": {"type": "string", "description": "Text to echo back"},
+                    },
+                    "required": ["text"],
+                },
+            ),
+        ]
+
+    @server.call_tool()
+    async def call_tool(name: str, arguments: dict[str, Any] | None) -> list[TextContent]:
+        """Handle tool calls with Unicode content."""
+        if name == "echo_unicode":
+            text = arguments.get("text", "") if arguments else ""
+            return [
+                TextContent(
+                    type="text",
+                    text=f"Echo: {text}",
+                )
+            ]
+        else:
+            raise ValueError(f"Unknown tool: {name}")
+
+    @server.list_prompts()
+    async def list_prompts() -> list[types.Prompt]:
+        """List prompts with Unicode names and descriptions."""
+        return [
+            types.Prompt(
+                name="unicode_prompt",
+                description="Unicode prompt - Слой хранилища, где располагаются",
+                arguments=[],
+            )
+        ]
+
+    @server.get_prompt()
+    async def get_prompt(name: str, arguments: dict[str, Any] | None) -> types.GetPromptResult:
+        """Get a prompt with Unicode content."""
+        if name == "unicode_prompt":
+            return types.GetPromptResult(
+                messages=[
+                    types.PromptMessage(
+                        role="user",
+                        content=types.TextContent(
+                            type="text",
+                            text="Hello世界🌍Привет안녕مرحباשלום",
+                        ),
+                    )
+                ]
+            )
+        raise ValueError(f"Unknown prompt: {name}")
+
+    # Create the session manager
+    session_manager = StreamableHTTPSessionManager(
+        app=server,
+        json_response=False,  # Use SSE for testing
+    )
+
+    @asynccontextmanager
+    async def lifespan(app: Starlette) -> AsyncGenerator[None, None]:
+        async with session_manager.run():
+            yield
+
+    # Create an ASGI application
+    app = Starlette(
+        debug=True,
+        routes=[
+            Mount("/mcp", app=session_manager.handle_request),
+        ],
+        lifespan=lifespan,
+    )
+
+    # Run the server
+    config = uvicorn.Config(
+        app=app,
+        host="127.0.0.1",
+        port=port,
+        log_level="error",
+    )
+    uvicorn_server = uvicorn.Server(config)
+    uvicorn_server.run()
+
+
+@pytest.fixture
+def unicode_server_port() -> int:
+    """Find an available port for the Unicode test server."""
+    import socket
+
+    with socket.socket() as s:
+        s.bind(("127.0.0.1", 0))
+        return s.getsockname()[1]
+
+
+@pytest.fixture
+def running_unicode_server(unicode_server_port: int) -> Generator[str, None, None]:
+    """Start a Unicode test server in a separate process."""
+    import multiprocessing
+    import socket
+    import time
```

Comment:
nit: I think most of these imports could be moved to the top of the file (I see the note on `run_unicode_server` though), we seem to import the same packages multiple times